apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -87,14 +144,98 @@ Keep in mind you are not just generating an example; you are generating an actua
       user: openAiUserFromAuthn(authnUserId),
     });
 
-    const completionText = completion.choices[0].message.content;
+    extractFromCompletion(completion, job);
+    job.data['prompt'] = prompt;
+    job.data['generation'] = completion.choices[0].message.content;
+    job.data['context'] = context;
+  });
 
-    job.info(`completion is ${completionText}`);
+  return { jobSequenceId: serverJob.jobSequenceId, jobData };
+}
 
-    job.info(`used ${completion?.usage?.total_tokens} OpenAI tokens to generate response.`);
+/**
+ * Revises a question using the LLM based on user input.
+ *
+ * @param client The OpenAI client to use.
+ * @param courseId The ID of the current course.
+ * @param authnUserId The authenticated user's ID.
+ * @param originalPrompt The prompt creating the original generation.
+ * @param revisionPrompt A prompt with user instructions on how to revise the question.
+ * @param originalHTML The question.html file to revise.
+ * @param originalPython The server.py file to revise.
+ * @returns A server job ID for the generation task and a promise to return the associated saved data on completion.
+ */
+export async function regenerateQuestion(
+  client: OpenAI,
+  courseId: string | undefined,
+  authnUserId: string,
+  originalPrompt: string,
+  revisionPrompt: string,
+  originalHTML: string,
+  originalPython: string,
+): Promise<{ jobSequenceId: string; jobData: ServerJobResult }> {
+  const serverJob = await createServerJob({
+    courseId,
+    type: 'llm_question_regen',
+    description: 'Revise a question using the LLM',
+  });
+
+  const jobData = await serverJob.execute(async (job) => {
+    job.info(`prompt is ${revisionPrompt}`);
+
+    const context = await makeContext(client, originalPrompt, authnUserId);
+    const sysPrompt = `
+        ${promptPreamble(context)}
```

Comment:
Same comment as above, consider dedenting this.

Reply:
done in e0331beaf9d587990dcda087d800feedc786fc87.